verifiers/internal/gha/slsaprovenance/v1.0/byob.go

```diff
@@ -0,0 +1,265 @@
+	}
+	ref, err := getValidateKey(workflowMap, "ref")
+	if err != nil {
+		return "", "", "", fmt.Errorf("%w: %v", serrors.ErrorMalformedURI, err)
+	}
+	repository, err := getValidateKey(workflowMap, "repository")
+	if err != nil {
+		return "", "", "", fmt.Errorf("%w: %v", serrors.ErrorMalformedURI, err)
+	}
+	path, err := getValidateKey(workflowMap, "path")
+	if err != nil {
+		return "", "", "", err
+	}
+	return repository, ref, path, nil
+}
+
+func (p *BYOBProvenanceV1) builderTriggerInfo() (string, string, string, error) {
+	sysParams, ok := p.prov.Predicate.BuildDefinition.InternalParameters.(map[string]interface{})
+	if !ok {
+		return "", "", "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "internal parameters type")
+	}
+
+	if _, exists := sysParams["GITHUB_WORKFLOW_REF"]; !exists {
+		return "", "", "", fmt.Errorf("%w: GITHUB_WORKFLOW_REF", serrors.ErrorNotPresent)
+	}
+
+	workflowRef, err := common.GetAsString(sysParams, "GITHUB_WORKFLOW_REF")
+	if err != nil {
+		return "", "", "", err
+	}
+
+	parts := strings.Split(workflowRef, "@")
+	if len(parts) != 2 {
+		return "", "", "", fmt.Errorf("%w: ref: %s", serrors.ErrorInvalidFormat, workflowRef)
+	}
+	repoAndPath := parts[0]
+	ref := parts[1]
+
+	parts = strings.Split(repoAndPath, "/")
+	if len(parts) < 2 {
+		return "", "", "", fmt.Errorf("%w: rep and path: %s", serrors.ErrorInvalidFormat, repoAndPath)
+	}
+
+	repo := strings.Join(parts[:2], "/")
+	path := strings.Join(parts[2:], "/")
+	return fmt.Sprintf("git+https://github.com/%s", repo), ref, path, nil
+}
+
+func (p *BYOBProvenanceV1) triggerInfo() (string, string, string, error) {
+	// TODO(#613): Support for generators.
+	return p.builderTriggerInfo()
+}
+
+// TriggerURI implements Provenance.TriggerURI.
+func (p *BYOBProvenanceV1) TriggerURI() (string, error) {
+	repository, ref, _, err := p.triggerInfo()
+	if err != nil {
+		return "", err
+	}
+	if repository == "" || ref == "" {
+		return "", fmt.Errorf("%w: repository or ref is empty", serrors.ErrorMalformedURI)
+	}
+	return fmt.Sprintf("%s@%s", repository, ref), nil
+}
+
+// Subjects implements Provenance.Subjects.
+func (p *BYOBProvenanceV1) Subjects() ([]intoto.Subject, error) {
+	subj := p.prov.Subject
+	if len(subj) == 0 {
+		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "no subjects")
+	}
+	return subj, nil
+}
+
+// GetBranch implements Provenance.GetBranch.
+func (p *BYOBProvenanceV1) GetBranch() (string, error) {
+	// Returns the branch from the source URI.
+	sourceURI, err := p.SourceURI()
+	if err != nil {
+		// Get the value from the internalParameters if there is no source URI.
+		sysParams, ok := p.prov.Predicate.BuildDefinition.InternalParameters.(map[string]interface{})
+		if !ok {
+			return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "internal parameters type")
+		}
+		return common.GetBranch(sysParams, slsa1.PredicateSLSAProvenance)
+	}
```

Comment:
NOTE that I fall back to the previous behavior of getting the branch from the `internalParameters`.

Some tests seem to have test data that doesn't include a `resolvedDependencies` so this is here to make those work but I'm not sure that there are real cases where provenance wouldn't have it.